compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -275,7 +275,7 @@ public BType checkType(Location pos,
     public boolean isLax(BType type) {
         Set<BType> visited = new HashSet<>();
         int result = isLaxType(type, visited);
-        if (result == 1) {
+        if (result == 1 || type.tag == TypeTags.XML || type.tag == TypeTags.XML_ELEMENT) {
```

Comment:
This seems wrong. this will return XML as lax, which is wrong.

Reply:
The `isLax` function is used when checking for field based access, and since XML maps won't give 1 for `isLaxType` now only XML attribute access will be allowed. Maybe the function name `isLax` is misleading here and can be changed to something like `isLaxFieldAccessAllowed` ?

Reply:
Please rename this.